File: solutions/cpp/1493-longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int left = 0, count_zeros = 0, max_length = 0;
        for (int right = 0; right < nums.size(); ++right) {
            if (nums[right] == 0) {
                ++count_zeros;
            }
            while (count_zeros > 1) {
                if (nums[left] == 0) {
                    --count_zeros;
                }
                ++left;
            }
            max_length = max(max_length, right - left);
        }
        return max_length;
    }
};

File: solutions/python/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        left, count_zeros, max_length = 0, 0, 0 
        for right in range(len(nums)): 
            if nums[right] == 0: 
                count_zeros += 1 
            while count_zeros > 1: 
                if nums[left] == 0: 
                    count_zeros -= 1 
                left += 1 
            max_length = max(max_length, right-left) 
        return max_length 
                                                                                                                                              
